Automattic/jetpack
# Social: Convert OG image from post body

Fixes the OpenGraph issue mentioned in pdrWKz-1rf-p2.

## Proposed changes:

* With Publicize enabled, if we have an OG image that's coming from the post body, we check it's filesize and if it's bigger than 4Mb-s we reduce it gradually to match the OG restrictions.

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/176
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* On trunk, have Publicize enabled and a Facebook account connected. Create a post with [this](https://href.li/?https://assets.sonos.com/media-kits/hardware/era-300/era-300-hero-horizontal-white.jpg) image in its body
* Publish the post. Check that on Facebook the link preview is not working. You can also copy the posts URL into this tool, and see that no preview has the image https://www.opengraph.xyz/

Check out this PR.
* Create another post with the same image as above, and share it.
* You should see the OG image on facebook in the link preview, and also in the tool I linked above
